lite/core/optimizer/mir/fusion/transformer_attention_fuse_pass.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lite/core/optimizer/mir/fusion/transformer_attention_fuse_pass.h"
+#include <list>
+#include <memory>
+#include <vector>
+#include "lite/core/optimizer/mir/fusion/transformer_attention_fuser.h"
+#include "lite/core/optimizer/mir/pass_registry.h"
+
+namespace paddle {
+namespace lite {
+namespace mir {
+
+void TransformerAttentionFusePass::Apply(
+    const std::unique_ptr<SSAGraph>& graph) {
+  fusion::TransformerAttentionFuser fuser;
+  bool has_int8 = false;
+  bool has_arm = false;
+  bool has_opencl = false;
+  for (auto& place : graph->valid_places()) {
+    if (place.precision == PRECISION(kInt8)) {
+      has_int8 = true;
+    }
+    if (place.target == TARGET(kARM)) {
+      has_arm = true;
+    }
+    if (place.target == TARGET(kOpenCL)) {
+      has_opencl = true;
+    }
+  }
+  if ((has_arm && has_int8)) {
+    fuser(graph.get());
+  } else {
+    return;
+  }
+}
+
+}  // namespace mir
+}  // namespace lite
+}  // namespace paddle
+
+REGISTER_MIR_PASS(lite_transformer_attention_fuse_pass,
+                  paddle::lite::mir::TransformerAttentionFusePass)
+    .BindTargets({TARGET(kOpenCL), TARGET(kARM)})
+    .BindKernel("fused_attention");
```

Comment:
```suggestion
    .ExcludeTargets({TARGET(kAny)})
    .BindTargets({TARGET(kARM)})
    .BindKernel("fused_attention");
```
- 看起来还没有支持 opencl
- kARM 是一种 host place，我们希望比如 xpu + arm 的情况下不进行 fuse

Reply:
是的，opencl相关支持还没有提交，另外的pr提交

Reply:
done